src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/commands/hmi/unsubscribe_button_request_test.cc

```diff
@@ -0,0 +1,140 @@
+typedef std::shared_ptr<UnsubscribeButtonRequest> UnsubscribeButtonRequestPtr;
+
+namespace {
+const uint32_t kCorrelationId = 2u;
+const uint32_t kAppId = 1u;
+const hmi_apis::Common_ButtonName::eType kButtonName =
+    hmi_apis::Common_ButtonName::CUSTOM_BUTTON;
+const hmi_apis::FunctionID::eType kFunctionID =
+    hmi_apis::FunctionID::Buttons_UnsubscribeButton;
+}  // namespace
+
+class HMIUnsubscribeButtonRequestTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+ protected:
+  MessageSharedPtr CreateCommandMsg() {
+    MessageSharedPtr command_msg(CreateMessage(smart_objects::SmartType_Map));
+    (*command_msg)[strings::msg_params][strings::app_id] = kAppId;
+    (*command_msg)[strings::msg_params][strings::button_name] = kButtonName;
+    (*command_msg)[strings::params][strings::correlation_id] = kCorrelationId;
+    (*command_msg)[strings::params][strings::function_id] = kFunctionID;
+
+    return command_msg;
+  }
+
+  void InitCommand(const uint32_t& timeout) OVERRIDE {
+    mock_app = CreateMockApp();
+    CommandRequestTest<CommandsTestMocks::kIsNice>::InitCommand(timeout);
+    ON_CALL((*mock_app), hmi_app_id()).WillByDefault(Return(kAppId));
+    ON_CALL(app_mngr_, application_by_hmi_app(kAppId))
+        .WillByDefault(Return(mock_app));
+    ON_CALL(app_mngr_, application(kAppId)).WillByDefault(Return(mock_app));
+  }
+
+  MockAppPtr mock_app;
+};
+
+TEST_F(HMIUnsubscribeButtonRequestTest, Run_SendRequest) {
+  MessageSharedPtr command_msg = CreateCommandMsg();
+  RequestToHMIPtr command(CreateCommand<UnsubscribeButtonRequest>(command_msg));
+  EXPECT_CALL(mock_rpc_service_, SendMessageToHMI(command_msg));
+  ASSERT_TRUE(command->Init());
+
+  command->Run();
+
+  EXPECT_EQ(CommandImpl::hmi_protocol_type_,
+            (*command_msg)[strings::params][strings::protocol_type].asInt());
+  EXPECT_EQ(CommandImpl::protocol_version_,
+            (*command_msg)[strings::params][strings::protocol_version].asInt());
+}
+
+TEST_F(HMIUnsubscribeButtonRequestTest, onTimeOut) {
+  MessageSharedPtr command_msg = CreateCommandMsg();
+  RequestToHMIPtr command(CreateCommand<UnsubscribeButtonRequest>(command_msg));
+
+  resumption_test::MockResumeCtrl mock_resume_ctrl;
+  EXPECT_CALL(app_mngr_, resume_controller())
+      .WillOnce(ReturnRef(mock_resume_ctrl));
+  EXPECT_CALL(mock_resume_ctrl, HandleOnTimeOut(_, _));
+
+  command->onTimeOut();
+}
+
+TEST_F(HMIUnsubscribeButtonRequestTest, on_event_SUCCESS) {
+  MessageSharedPtr command_msg = CreateCommandMsg();
+
+  UnsubscribeButtonRequestPtr command =
+      CreateCommand<UnsubscribeButtonRequest>(command_msg);
+
+  MessageSharedPtr event_msg = CreateCommandMsg();
+  (*event_msg)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::SUCCESS;
+  Event event(kFunctionID);
+  event.set_smart_object(*event_msg);
+
+  EXPECT_CALL(*mock_app, UnsubscribeFromButton(_)).WillOnce(Return(true));
+  command->on_event(event);
+}
+
```

Comment:
Please add negative case where result is not successful and we don't unsubscribe from button